tests/mapping/test_mapping_script.py

```diff
@@ -0,0 +1,121 @@
+import unittest
+import rdflib
+import datetime
+import os
+import yaml
+from pathlib import Path
+
+from lebedigital.mapping.emodul_mapping import generate_knowledge_graph, get_date_time_value
+
+class TestMappingScript(unittest.TestCase):
+
+    def query_and_test_result(self, typeOfQuery, predicate, prop, g):
+        print(f"""Query for {typeOfQuery}:""")
+        q = f"""
+                SELECT ?o WHERE {{
+                        {predicate} {prop} ?o .
+                }}
+                """
+        print(q)
+        result = g.query(q)
+        return result
+
+
+    def print_next(self):
+        print("Pass")
+        print("###################Next Query#####################################")
+
+
+    def print_first(self):
+        print("###################First Query####################################")
+
+
+    def test_generate_knowledge_graph(self):
+        metadataPath = 'testMetaData.yaml'
+        filename = "knowledgeGraph.ttl"
+        print("Start testing:")
+        # Generate a testMetadata file
+        print("Generate metadata file for test:")
+        target_data = {'experimentName': 'BA-Losert MI E-Modul 28d v. 04.08.14 Probe 4',
+                       'software_specification': 'MTS793|MPT|DEU|1|2|,|.|:|49|1|1|A',
+                       'operator_timestamp': '13:25:39',
+                       'operator_date': '01.09.2014',
+                       'tester_name': 'Kh',
+                       'specimen_name': 'BA-Losert E-Modul 28d v. 04.08.14 Probe 4',
+                       'remark': 'Kraftgeregelt 3,9 kN/s',
+                       'weight': 5342.0,
+                       'weight_unit': 'g',
+                       'diameter': 98.6,
+                       'length': 300.3,
+                       'length_unit': 'mm',
+                       'mix_file': '2014_08_05 Rezeptur_MI.xlsx'}
+
+        with open(metadataPath, 'w') as yamlFile:
+            yaml.dump(target_data, yamlFile)
+        print("Generate knowledgeGraph:")
+        generate_knowledge_graph(metadataPath, filename)
+
+        g = rdflib.Graph()
+        result = g.parse(filename, format='ttl')
+        print("Query knowledgeGraph:")
+
+        self.print_first()
+
+        result = self.query_and_test_result('diameter', 'ns1:informationbearingentity1',
```

Comment:
This only checks of there is a single informationbearingentity1 with the value. How do we know that this is the diameter? I would think it might be relevant to start from the id of the test and then extract its diameter traversing the complete graph. This might look redundant (and does not allow to have a query_and_test method) but IMO advantageous (since this is how we are supposed to use the knowledge graph afterwards). This is true for all the subsequent tests.

Reply:
Could you please give me more info related this issue? How the function should change in order to cover the new requirements? It seems a bit confusing, do you mean it's needed to change the query?

Reply:
yes, the query is supposed to search in a complete triple store with all test (potentially multiple test each having a diameter)

Reply:
@joergfunger are you sure? I thought this test is supposed to check the mapping of extracted data to ttl file.
The connection to the triple store is one or two steps further, right?
This is still an open issue but not related to this test.

Reply:
The data structure that is generated is a graph. We are currently just checking the last leaf of the graph, without any understanding of the connections. IMO, the test should traverse the complete graph, such that this graph is tested, not only the final leaf. Thus, the question to be answered would be to return the diameter of a Young's modulus test (with a given id) by traversing from the test to the specimen and then the diameter.